### PR TITLE
[IMP] developer/view_architectures: new date filters

### PR DESCRIPTION
**Pull request by MissingNoShiny**
Community: https://github.com/odoo/odoo/pull/156746
Enterprise: https://github.com/odoo/enterprise/pull/59901
Upgrade: https://github.com/odoo/upgrade/pull/6034

Task-3692123

### CODE REVIEW

**Review by Feyensv — APPROVED** [review]
@robodoo delegate+

PS: when doing commits in other odoo repositories, the commit target should ideally be adapted to the effective changes (& modified files) in the repository.

**Comment by MissingNoShiny** [comment]
Sorry, I should have opened this PR in draft, I will have to make further changes in the community branch, and reflect them in the documentation

**Comment by MissingNoShiny** [comment]
Hello @Feyensv , this PR is ready for a final review now :slightly_smiling_face:

File: content/developer/reference/user_interface/view_architectures.rst [hunk 1]
```diff
@@ -1917,28 +1917,121 @@ The `filter` element can have the following attributes:
...
+   :noindex:
+
+   The earliest month that will show up in the dropdown of a date filter, as an offset relative to the current month.
+   For instance, if the current month is February, a `start_month` with a value of -3 means that the earliest month
+   in the dropdown will be November.
+
+   .. example::
+      .. code-block:: xml
+
+         <filter string="Creation Date" name="filter_create_date" date="create_date" start_month="-3"/>
```
Comment (Feyensv) [inline content/developer/reference/user_interface/view_architectures.rst:1947]: IMHO, your textual examples could be moved into the `.. example` sections, to keep the attribute definition short and simple.

```suggestion
   .. example::
      .. code-block:: xml

         <filter string="Creation Date" name="filter_create_date" date="create_date" start_month="-3"/>
         
       If the current month is February, the earliest month shown in the dropdown will be November.
```

**Review by Feyensv — APPROVED** [review]
LGTM

@robodoo delegate+
(already done previously, but it's clear to say it again :D)

PS: why do you reference your PRs in your commit message ?  The mergebot will already automatically add those references, and it creates a loooot of noise in your PRs.

**Pull request closed without merging**